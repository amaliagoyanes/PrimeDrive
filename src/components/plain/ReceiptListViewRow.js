import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {toFixedLocale, toFixedLocaleDa} from './receipts/plainComponents'
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling';

export default class ReceiptListViewRow extends React.Component {
    render() {
        let ticketLine = this.props.data;
        let discount = '';
        let amount = '';
        let text = '';
        let splitted = (ticketLine.splitUnits>0) ? ticketLine.splitRatio : 1;

        if (ticketLine && ticketLine.discount !== undefined && ticketLine.discount.length)
            discount = toFixedLocaleDa(ticketLine.product_price * ticketLine.units * (parseFloat(ticketLine.discount.replace(',', '.')) * splitted / 100));

        if (ticketLine && ticketLine.campaign !== undefined && ticketLine.campaign.amount && ticketLine.campaign.type === 'discount') {
            amount = toFixedLocaleDa(ticketLine.product_price * ticketLine.units * (parseFloat(ticketLine.campaign.amount.replace(',', '.')) * splitted / 100));
            text = `${this.props.intl.messages["app.campaign"]} ${ticketLine.campaign.name} -${ticketLine.campaign.amount}%`;
        }
        if (ticketLine && ticketLine.campaign !== undefined && ticketLine.campaign.amount && ticketLine.campaign.type === 'price') {
            amount = toFixedLocaleDa(ticketLine.campaign.amount.replace(',', '.') * ticketLine.units * splitted);
            text = `${this.props.intl.messages["app.campaign"]} ${ticketLine.campaign.name}: ${ticketLine.campaign.amount} DKK`;
        }

        return (
            <Row style={{paddingTop: scale(1), paddingBottom: scale(5)}}><Col>
                <Row>
                    <Text style={[styles.text, styles.sectionReceiptProduct, {width: '50%'}]}>
                        {ticketLine.name}
                    </Text>
                    <Text style={[styles.text, styles.sectionReceiptProduct, {width: '25%', textAlign: 'right'}]}>
                        {ticketLine.units}{(ticketLine.splitRatio)? `x${Number(ticketLine.splitRatio).toFixed(2)}`:''} x
                    </Text>
                    <Text style={[styles.text, styles.sectionReceiptProductPrice, {width: '25%'}]}>
                        {toFixedLocaleDa(ticketLine.product_price * ticketLine.units * splitted)}
                    </Text>
                </Row>
                {ticketLine.campaign !== undefined && !!ticketLine.campaign.type.length && parseFloat(ticketLine.campaign.amount.replace(',', '.')) > 0 && 
                <Row>
                    <Text style={[styles.text, styles.sectionReceiptCampaign]}>{text}</Text>
                    <Text style={[styles.text, styles.sectionReceiptCampaignAmount]}>{ticketLine.campaign.type==='discount' && !(/-/.test(amount))?'-':''}{amount} DKK</Text>
                </Row>
                }
                {ticketLine.discount !== undefined && parseFloat(ticketLine.discount.replace(',', '.')) > 0 &&
                <Row>
                    <Text style={[styles.text, styles.sectionReceiptCampaign]}>
                        {this.props.intl !== undefined && this.props.intl.messages["app.discount"]}</Text>
                    <Text style={[styles.text, styles.sectionReceiptCampaignAmount]}>
                        {(/-/.test(discount))?'':'-'}{discount}</Text>
                </Row>
                }
            </Col></Row>
        );
    }
}