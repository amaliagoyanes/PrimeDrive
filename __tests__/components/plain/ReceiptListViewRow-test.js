'use strict';

import 'react-native';
import React from 'react';
import ReceiptListViewRow from '../../../src/components/plain/ReceiptListViewRow';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let data = {name: 'Test', units: 1, product_price: 5};
it('renders correctly', () => {
    const tree = renderer.create(
        <ReceiptListViewRow data={data} onDelete={jest.fn} onPress={jest.fn}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

let data2 = {name: 'Test', units: 1, product_price: 5, discount: '20'};
it('renders correctly with discount', () => {
    const tree = renderer.create(
        <ReceiptListViewRow data={data2} onDelete={jest.fn} onPress={jest.fn}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
