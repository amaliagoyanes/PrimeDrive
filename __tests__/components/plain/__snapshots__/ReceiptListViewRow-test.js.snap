// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingBottom": 1,
      "paddingTop": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Gotham-Book",
                "fontSize": 14,
              },
              Object {
                "color": "#afafaf",
                "fontSize": 9,
                "width": "60%",
              },
            ]
          }
        >
          Test
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Gotham-Book",
                "fontSize": 14,
              },
              Object {
                "color": "#afafaf",
                "fontSize": 9,
                "textAlign": "right",
                "width": "40%",
              },
            ]
          }
        >
          5,00
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with discount 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingBottom": 1,
      "paddingTop": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Gotham-Book",
                "fontSize": 14,
              },
              Object {
                "color": "#afafaf",
                "fontSize": 9,
                "width": "60%",
              },
            ]
          }
        >
          Test
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Gotham-Book",
                "fontSize": 14,
              },
              Object {
                "color": "#afafaf",
                "fontSize": 9,
                "textAlign": "right",
                "width": "40%",
              },
            ]
          }
        >
          5,00
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Gotham-Book",
              "fontSize": 14,
            },
            Object {
              "color": "#afafaf",
              "fontSize": 9,
              "width": "60%",
            },
          ]
        }
      >
        Discount
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Gotham-Book",
              "fontSize": 14,
            },
            Object {
              "color": "#afafaf",
              "fontSize": 9,
              "textAlign": "right",
              "width": "40%",
            },
          ]
        }
      >
        -
        1,00
      </Text>
    </View>
  </View>
</View>
`;
